src/inmanta/data.py

```diff
@@ -1130,34 +1130,62 @@ class Report(BaseDocument):
         :param outstream what was reported on system out
     """
 
-    id = Field(field_type=uuid.UUID, required=True, part_of_primary_key=True)
-    started = Field(field_type=datetime.datetime, required=True)
-    completed = Field(field_type=datetime.datetime, required=True)
-    command = Field(field_type=str, required=True)
-    name = Field(field_type=str, required=True)
-    errstream = Field(field_type=str, default="")
-    outstream = Field(field_type=str, default="")
-    returncode = Field(field_type=int)
-    compile = Field(field_type=uuid.UUID)
+    id: uuid.UUID = Field(field_type=uuid.UUID, required=True, part_of_primary_key=True)
+    started: datetime.datetime = Field(field_type=datetime.datetime, required=True)
+    completed: datetime.datetime = Field(field_type=datetime.datetime)
+    command: str = Field(field_type=str, required=True)
+    name: str = Field(field_type=str, required=True)
+    errstream: str = Field(field_type=str, default="")
+    outstream: str = Field(field_type=str, default="")
+    returncode: int = Field(field_type=int)
+    compile: uuid.UUID = Field(field_type=uuid.UUID, required=True)
+
+    async def update_streams(self, out="", err="") -> None:
+        if not out and not err:
+            return
+        await self._execute_query(
+            f"UPDATE {self.table_name()} SET outstream = outstream || $1, errstream = errstream || $2 WHERE id = $3",
+            self._get_value(out),
+            self._get_value(err),
+            self._get_value(self.id),
+        )
 
 
 class Compile(BaseDocument):
     """
         A run of the compiler
 
-        :param environment The environment this resource is defined in
-        :param started Time the compile started
-        :param completed Time to compile was completed
-        :param reports Per stage reports
+        :param environment: The environment this resource is defined in
+        :param requested: Time the compile was requested
+        :param started: Time the compile started
+        :param completed: Time to compile was completed
+        :param do_export: should this compiler perform an export
+        :param force_update: should this compile definitely update
+        :param metadata: exporter metadata to be passed to the compiler
+        :param environment_variable: environment variables to be passed to the compiler
+        :param succes: was the compile successful
+        :param handled: were all registered handlers executed?
+        :param version: version exported by this compile
     """
 
-    id = Field(field_type=uuid.UUID, required=True, part_of_primary_key=True)
-    environment = Field(field_type=uuid.UUID, required=True)
-    started = Field(field_type=datetime.datetime)
-    completed = Field(field_type=datetime.datetime)
+    id: uuid.UUID = Field(field_type=uuid.UUID, required=True, part_of_primary_key=True)
+    remote_id: uuid.UUID = Field(field_type=uuid.UUID)
```

Comment:
if its not required should we make it Optional[uuid.UUID] and assign None? And same for other fields